halotools/empirical_models/abunmatch/engines/bin_free_cam_kernel.pyx

```diff
@@ -159,7 +183,9 @@ def cython_bin_free_cam_kernel(double[:] y1, double[:] y2, int[:] i2_match, int
     cdef int idx_in1, idx_out1, idx_in2, idx_out2
     cdef double value_in1, value_out1, value_in2, value_out2
 
-    cdef double[:] y1_new = np.zeros(npts1, dtype='f8') - 1
+    cdef int[:] y1_new_indexes = np.zeros(npts1, dtype='i4') - 1
```

Comment:
I tried to define these in if statements:
```
if return_indexes:
    cdef int[:] y1_new_indexes = np.zeros(npts1, dtype='i4') - 1
else:
    cdef double[:] y1_new_values = np.zeros(npts1, dtype='f8') - 1
```
but that wasn't allowed?

Defining both is a bit unnecessary, but I couldn't find a nice way around it

Reply:
Actually, I don't know a way around this either. This seems fine.